sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -115,6 +115,30 @@ void gnrc_ipv6_nib_init(void)
 
 static void _add_static_lladdr(gnrc_netif_t *netif)
 {
+#if defined(CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK) && \
+    CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK > 0
+#ifndef MODULE_NETDEV_REGISTER
+#error "Use of CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK requires MODULE_NETDEV_REGISTER"
+#endif
+    if (! ((CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK & (1ULL << NETDEV_ANY)) ||
```

Comment:
`NETDEV_ANY` should always match, but you can leave it out as that case makes no sense - it's the same as leaving `CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK`  as 0

Reply:
Would you agree to altering the guard of this code to

```C
#if (CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK > 0) && \
        ! ((CONFIG_GNRC_IPV6_STATIC_LLADDR_NETDEV_MASK & (1ULL << NETDEV_ANY))
```

rather than including this check in the if-statement? `NETDEV_ANY` is defined, so I would like to avoid leaving it uncovered. This might be clearer. I was just expecting that the compiler should be able to optimize the code away if `NETDEV_ANY` is checked in the if statement.

Reply:
Sounds good!

Reply:
The above suggestion works on a small test program using gcc but somehow not when compiling in RIOT. It does not throw an error but it always results in not including the code. I therefore opted to leave the `NETDEV_ANY` selection rule in the if statement and would hope for the compiler to optimize it away.